runtime-attach/src/main/java/io/opentelemetry/contrib/attach/RuntimeAttach.java

```diff
@@ -7,22 +7,82 @@
 
 import java.io.File;
 import java.lang.management.ManagementFactory;
+import java.util.logging.Logger;
 import net.bytebuddy.agent.ByteBuddyAgent;
 
+/** This class allows you to attach the OTel agent for Java at runtime. */
 public final class RuntimeAttach {
 
+  private static final Logger LOGGER = Logger.getLogger(RuntimeAttach.class.getName());
+  private static final String AGENT_ENABLED_PROPERTY = "otel.javaagent.enabled";
+  private static final String AGENT_ENABLED_ENV_VAR = "OTEL_JAVAAGENT_ENABLED";
+  static final String MAIN_THREAD_CHECK_PROP = "otel.javaagent.runtimeattach.mainthreadcheck";
+
+  /**
+   * Attach the OTel agent for Java to the current JVM. The attachment must be requested at the
+   * beginning of the main method.
+   */
   public static void attachJavaagentToCurrentJVM() {
-    if (agentIsDisabled()) {
+    if (!shouldAttach()) {
       return;
     }
+
     File javaagentFile = AgentFileLocator.locateAgentFile();
     ByteBuddyAgent.attach(javaagentFile, getPid());
+
+    if (!agentIsAttached()) {
+      LOGGER.warning("Agent was not attached. An unexpected issue has happened.");
+    }
+  }
+
+  private static boolean shouldAttach() {
+    if (agentIsDisabledWithProp()) {
+      LOGGER.warning("Agent was disabled with " + AGENT_ENABLED_PROPERTY + " property.");
+      return false;
+    }
+    if (agentIsDisabledWithEnvVar()) {
+      LOGGER.warning("Agent was disabled with " + AGENT_ENABLED_ENV_VAR + " environment variable.");
+      return false;
+    }
```

Comment:
do we need to check the env var and system property? it may be ok for the runtime attach component to try to attach and the javaagent will disable itself if these are set

Reply:
I understand your point of view @trask . It could make sense.  I would prefer not to try to attach, in such a case, to stop the things sooner. It's the [current behavior](https://github.com/open-telemetry/opentelemetry-java-contrib/commit/858dbae93b80081ebd47261f5dcb1b4abe387a04). What do you think @iNikem?

Reply:
ah, I hadn't noticed this was the current behavior, let's keep it then, thx